gtda/homology/grids.py

```diff
@@ -113,6 +127,15 @@ def fit(self, X, y=None):
         self : object
 
         """
+        self._filtration_kwargs = {}
+        if self.periodic_dimensions is None or \
+           np.sum(self.periodic_dimensions) == 0:
+            self.filtration = CubicalComplex
+        else:
+            self.filtration = PeriodicCubicalComplex
+            self._filtration_kwargs['periodic_dimensions'] = \
+                self.periodic_dimensions
+
         if self.infinity_values is None:
             self.infinity_values_ = np.max(X)
```

Comment:
What's the logic for having this as ``np.max(X)``? I believe there is no counterpart of this for ``VietorisRipsPersistence`` for example.

Reply:
What do you suggest? Which value would make more sense than taking the maximum value withh all grayscale image for an infinite value?

Reply:
I think I was getting worried because here there is no counterpart to `max_edge_length`. Perhaps the strict analogy to `VietorisRipsPersistence` would lead to choosing `np.inf`?

Reply:
IMO `np.inf` cannot be a default value because it makes future transformers hard to use if diagrams include infinite death value (they can't be sampled).

In the case of `VietorisRips`, the default parameters do not lead to that problem.